extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/storage/sql/SQLCatalogManager.java

```diff
@@ -0,0 +1,578 @@
+                  .bind("updateTime", updateTime)
+                  .bind("state", TableMetadata.TableState.ACTIVE.code())
+                  .bind("payload", table.spec().toBytes(jsonMapper));
+              try {
+                stmt.execute();
+              }
+              catch (UnableToExecuteStatementException e) {
+                if (DbUtils.isDuplicateRecordException(e)) {
+                  throw new DuplicateKeyException(
+                        "Tried to insert a duplicate table: " + table.sqlName(),
+                        e);
+                } else {
+                  throw e;
+                }
+              }
+              sendAddition(table, updateTime);
+              return updateTime;
+            }
+          }
+      );
+    }
+    catch (CallbackFailedException e) {
+      if (e.getCause() instanceof DuplicateKeyException) {
+        throw (DuplicateKeyException) e.getCause();
+      }
+      throw e;
+    }
+  }
+
+  @Override
+  public TableMetadata read(TableId id)
+  {
+    return dbi.withHandle(
+        new HandleCallback<TableMetadata>()
+        {
+          @Override
+          public TableMetadata withHandle(Handle handle)
+          {
+            Query<Map<String, Object>> query = handle.createQuery(
+                StringUtils.format(SELECT_TABLE, tableName)
+            )
+                .setFetchSize(connector.getStreamingFetchSize())
+                .bind("schemaName", id.schema())
+                .bind("name", id.name());
+            final ResultIterator<TableMetadata> resultIterator =
+                query.map((index, r, ctx) ->
+                  new TableMetadata(
+                      id,
+                      r.getLong(1),
+                      r.getLong(2),
+                      TableMetadata.TableState.fromCode(r.getString(3)),
+                      TableSpec.fromBytes(jsonMapper, r.getBytes(4))
+                  ))
+                .iterator();
+            if (resultIterator.hasNext()) {
+              return resultIterator.next();
+            }
+            return null;
+          }
+        }
+    );
+  }
+
+  @Override
+  public long update(TableMetadata table, long oldVersion) throws OutOfDateException, NotFoundException
+  {
+    if (oldVersion == 0) {
+      return updateUnsafe(table.id(), table.spec());
+    } else {
+      return updateSafe(table.id(), table.spec(), oldVersion);
+    }
+  }
+
+  private long updateSafe(TableId id, TableSpec defn, long oldVersion) throws OutOfDateException
+  {
+    try {
+      return dbi.withHandle(
+          new HandleCallback<Long>()
+          {
+            @Override
+            public Long withHandle(Handle handle) throws OutOfDateException
+            {
+              long updateTime = System.currentTimeMillis();
+              int updateCount = handle.createStatement(
+                  StringUtils.format(UPDATE_DEFN_SAFE, tableName))
+                  .bind("schemaName", id.schema())
```

Comment:
I dont believe that the schema or name of table can change can it? Since doing so would create a new entry in table. Given this maybe we shouldnt bind the schemaName nor name in the  update clause here. Same for other update methods.